### PR TITLE
feat(metrics): add run_id field and deduplicate by runId at report time

### PR DESCRIPTION
**Pull request by koydas**
metrics/runs.jsonl has no file locking, so two concurrent workflow runs
(e.g. auto-fix triggered during a pr-review) can both append records for
the same logical event.

Each appendMetric call now includes run_id = GITHUB_RUN_ID (unique per
Actions run) with a local-<epoch-ms> fallback for non-CI invocations.
metrics-report.mjs deduplicates at read time — first occurrence per
run_id wins; records without run_id (historical data) are kept as-is.
The report header reports the count of dropped duplicates when non-zero.

docs/runbook.md updated: run_id added to key-fields tables, the
"duplicate lines" troubleshooting row updated to reflect automatic
deduplication, and a backward-compat note added.

https://claude.ai/code/session_018aVCgNeGGrN4zS3oo8kEWx

### CODE REVIEW

**Comment by koydas** [comment]
## 🔍 Automated Code Review

### ✅ Summary
Adds `run_id` field to metrics records and deduplication logic in `metrics-report.mjs` to handle concurrent workflow runs. Updates docs/ADR/runbook to document deduplication strategy.

### ⚠️ Issues Found
- [High] Missing test for deduplication edge case — File: `scripts/tests/metrics.test.mjs` Lines: unknown  
  Root cause: Test only verifies basic deduplication but doesn't cover historical records without `run_id` or mixed CI/local `run_id` formats.  
  Fix: Add test case with 2+ records sharing same `run_id` (CI and local formats) plus 1 record missing `run_id` to verify all 3 are handled correctly.

### 🚀 Verdict
APPROVED

Automation gates satisfied:  
- ✅ Unit tests added for new deduplication logic  
- ✅ Docs updated in ADR, runbook, and key fields tables  
- ✅ Test coverage for core logic (minimum coverage not explicitly stated but existing tests validate critical path)

**Comment by koydas** [comment]
@Codex review

File: scripts/metrics-report.mjs [hunk 1]
```diff
@@ -4,13 +4,27 @@ import { readMetrics } from './lib/metrics.mjs';
...
-if (records.length === 0) {
+const seenRunIds = new Set();
+const deduped = [];
+let droppedCount = 0;
+for (const r of records) {
+  if (r.run_id == null) {
+    deduped.push(r);
+  } else if (!seenRunIds.has(r.run_id)) {
+    seenRunIds.add(r.run_id);
+    deduped.push(r);
```
[the remark this answers is not in the document]
Comment (koydas) [reply scripts/metrics-report.mjs]: Fixed in `555e0d6` — `run_id` now uses `CHECKPOINT_RUN_ID` (e.g. `issue-42`, `pr-15`) instead of `GITHUB_RUN_ID`. Since `CHECKPOINT_RUN_ID` is the same for all concurrent workflow runs processing the same issue or PR, two concurrent triggers of `validate-issue` for issue #42 both emit `run_id: "issue-42"` and are correctly deduplicated. `deduplicateMetrics` was also extracted as a tested exported function in `scripts/lib/metrics.mjs` with 6 test cases covering all branches.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by koydas** [comment]
@Codex review

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/lib/metrics.mjs [hunk 2]
```diff
@@ -28,3 +28,13 @@ export async function readMetrics() {
     throw err;
   }
 }
+
+export function deduplicateMetrics(records) {
+  const seen = new Set();
+  return records.filter((r) => {
+    if (r.run_id == null) return true;
+    if (seen.has(r.run_id)) return false;
```
[the remark this answers is not in the document]
Comment (koydas) [reply scripts/lib/metrics.mjs:36]: Correct — `CHECKPOINT_RUN_ID` est un identifiant logique stable (`issue-42`, `pr-15`) partagé par tous les runs du même issue/PR, ce qui causait la suppression silencieuse des re-validations et des records APPROVE postérieurs à un MANUAL. Reverted in `293481a` : `run_id` utilise à nouveau `GITHUB_RUN_ID` (unique par run Actions). La dédup dans `deduplicateMetrics` agit uniquement comme filet de sécurité same-run ; les doublons inter-runs distincts ne sont pas dédupliqués automatiquement, ce qui est documenté dans le runbook avec la procédure de nettoyage manuel.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (koydas) [reply scripts/lib/metrics.mjs:36]: Même point que le thread précédent — reverted in `293481a`. `run_id` est maintenant `GITHUB_RUN_ID` (unique par run) ; les records légitimes de runs successifs pour le même issue/PR sont tous conservés. Le runbook documente explicitement que la dédup ne couvre que les double-écritures du même run, et donne la procédure de nettoyage manuel pour les rares cas de doublons inter-runs.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by koydas** [comment]
@Codex review

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/lib/metrics.mjs (hunk 2 above)
[the remark this answers is not in the document]
Comment (koydas) [reply scripts/lib/metrics.mjs:36]: Fixed in `c889287` — `run_id` est maintenant `{GITHUB_RUN_ID}-{GITHUB_RUN_ATTEMPT}` (ex: `12345678-1`). GitHub garde `GITHUB_RUN_ID` constant sur les re-runs et incrémente `GITHUB_RUN_ATTEMPT`, donc chaque tentative produit un `run_id` distinct et n'est pas supprimée par `deduplicateMetrics`. Runbook et ADR-0013 mis à jour.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by koydas** [comment]
@Codex review

---
_Generated by [Claude Code](https://claude.ai/code)_
